templates/terraform/aws_agent_with_opentelemetry_collector/main.tf

```diff
@@ -37,11 +37,12 @@ module "agent" {
   source  = "monte-carlo-data/mcd-agent/aws"
   version = "1.0.3"
 
-  cloud_account_id  = var.cloud_account_id
-  private_subnets   = var.existing_subnet_ids
-  image             = var.agent_image_uri
-  region            = var.region
-  remote_upgradable = var.remote_upgradable
+  cloud_account_id        = var.cloud_account_id
+  private_subnets         = var.existing_subnet_ids
+  image                   = var.agent_image_uri
+  region                  = var.region
+  remote_upgradable       = var.remote_upgradable
+  create_s3_bucket_policy = !var.deploy_redshift_resources
```

Comment:
the `agent` module should manage the bucket policy when not deploying Redshift resources